### ISSUE TITLE
Investigate GraphQL Error: Variable "$onlyPublishedContent" of required type "Boolean!" was not provided.

### ISSUE THREAD

**Issue by ndouglas** [department-of-veterans-affairs/va.gov-cms#7835]
## Description
See e.g. in [Sentry](http://sentry.vfs.va.gov/organizations/vsp/issues/57419/?project=14&referrer=webhooks_plugin).

Can be reproduced with this query:

```graphql
  query($onlyPublishedContent: Boolean!) {
    
  outreachAssets: nodeQuery(filter: {conditions: [{field: "type", value: "outreach_asset", field: "status", value: ["1"], enabled: $onlyPublishedContent}]}, limit: 10000) {
    entities {
      ... on NodeOutreachAsset {
        entityId
        title
        status
        changed
        fieldFormat
        fieldBenefits
        fieldDescription
        fieldListing {
          targetId
        }
        fieldMedia {
          entity {
            ... on MediaImage {
              entityBundle
              image {
                entity{
                  filesize
                }
                url
                alt
              }
            }
            ... on MediaDocument {
              entityBundle
              fieldDocument {
                entity {
                  filesize
                  url
                }
              }
            }
            ... on MediaVideo {
              entityBundle
              fieldMediaVideoEmbedField
              thumbnail {
                derivative(style: LARGE) {
                  url
                }
              }
            }
          }
        }
      }
    }
  }

  }
```

I don't know enough GraphQL to know what exactly is going on here, but this error occurs a lot.

## Acceptance Criteria
- [ ] Reproduce issue and determine root cause
- [ ] Resolve issue or create followup ticket as needed

### CMS Team

Please leave only the team that will do this work selected. 

- [x] `Platform CMS Team`

**Comment by ElijahLynn** [comment]
FWIW: This error has also happened to me many times using the GraphQL Explorer tool by not providing the right variable at the bottom in the variable window. So "some" of these types of errors may be caused by that scenario, though it looks like not all of them. Just want to say if we fix this scenario with outreach that we may still see them.

**Comment by ElijahLynn** [comment]
If we pass in:

```
{
    "onlyPublishedContent": true
}
```

Then the error goes away when using GraphiQL Explorer. 
![image](https://user-images.githubusercontent.com/1504756/153489965-1889f8ff-7d35-44d9-a427-9c6b85790f19.png)

The errors in the logs are using the ELB address, not prod.cms.va.gov, so it does look like they are being caused by automated scripts, not the explorer:

![image](https://user-images.githubusercontent.com/1504756/153490111-430d3b89-d497-4895-bcda-a9f5ea0d19a8.png)

**Comment by timcosgrove** [comment]
Discoveries so far:

1. The queries generating this are almost certainly coming from a cron process. If you look at the queries in recent log, it very clearly is on a 15 minute cycle: https://prod.cms.va.gov/admin/reports/dblog?type%5B%5D=graphql
2. Correspondingly, these bad queries do not seem to be coming from content-build. Content build does not run on a set schedule like this, and we don't see instances where that steady every-15-minutes pattern is interrupted. 
3. There does not appear to be anything in vets-api that would be generating these queries.

**Comment by timcosgrove** [comment]
There are two IPs associated with these requests, visible in some of the errors:
https://prod.cms.va.gov/admin/reports/dblog/event/5834104
https://prod.cms.va.gov/admin/reports/dblog/event/5834733

These are not internal IPs that I can identify with my current level of AWS/dsvagovcloud access.

**Comment by timcosgrove** [comment]
@ndouglas did some investigation:

> Poked around in Athena and found that request:
client_ip: 10.247.32.148
user_agent: node-fetch/1.0 (+https://github.com/bitinn/node-fetch[](https://files.slack.com/files-pri/T03FECE8V-F034CM3U1N0/screen_shot_2022-02-23_at_8.02.29_am.png)[](https://files.slack.com/files-pri/T03FECE8V-F034CM3U1N0/download/screen_shot_2022-02-23_at_8.02.29_am.png))
That matches up fairly well with content build.
I found that IP's interface in the elastic IPs:
Screen Shot 2022-02-23 at 8.02.29 AM.png 
Screen Shot 2022-02-23 at 8.02.29 AM.png

> [Nathan Douglas](https://app.slack.com/team/U01DADWHLM6)  [4 hours ago](https://dsva.slack.com/archives/C02HX4AQZ33/p1645621445165669?thread_ts=1645576595.468499&cid=C02HX4AQZ33)
That leads here: https://console.amazonaws-us-gov.com/ec2/home?region=us-gov-west-1#InstanceDetails:instanceId=i-04e2eac55f48eb8c0[](https://files.slack.com/files-pri/T03FECE8V-F034ABRA8RG/screen_shot_2022-02-23_at_8.03.44_am.png)[](https://files.slack.com/files-pri/T03FECE8V-F034ABRA8RG/download/screen_shot_2022-02-23_at_8.03.44_am.png)
Screen Shot 2022-02-23 at 8.03.44 AM.png 
Screen Shot 2022-02-23 at 8.03.44 AM.png

> [Nathan Douglas](https://app.slack.com/team/U01DADWHLM6)  [4 hours ago](https://dsva.slack.com/archives/C02HX4AQZ33/p1645621457191189?thread_ts=1645576595.468499&cid=C02HX4AQZ33)
Looks like it's the prod content preview server

> [Nathan Douglas](https://app.slack.com/team/U01DADWHLM6)  [2 hours ago](https://dsva.slack.com/archives/C02HX4AQZ33/p1645630749991699?thread_ts=1645576595.468499&cid=C02HX4AQZ33)
Makes sense.  $onlyPublishedContent is normally true, because that's what the frontend wants.  The preview server will set that value to false because it needs to work with unpublished content.  :cool-doge:  So just need to find the code where the preview server passes in diddly-squat instead of passing an explicit false.

**Comment by timcosgrove** [comment]
This appears to be where the requests are originating: https://github.com/department-of-veterans-affairs/content-build/blob/master/script/preview.js#L362

**Comment by timcosgrove** [comment]
This is waiting on the next content build deploy, at which point we will re-deploy the preview server.
